subsys/settings/src/settings_ram.c

```diff
@@ -0,0 +1,270 @@
+
+/* the RAM settings backend uses static SList to store settings in heap */
+static sys_slist_t settings_list = SYS_SLIST_STATIC_INIT(&settings_list);
+
+/**
+ * @brief Try to update already existing entry
+ * 
+ * Search along the settings list and try to find existing entry.
+ * If entry exist, free its value and reserve the memory for a new entry.
+ * 
+ * @param name name of the entry / key
+ * @param value value to be stored in settings
+ * @param len length of the value
+ * @return true if entry has been stored to the setting list
+ * @return false if entry does not exist or there is no RAM space to store bigger value.
+ */
+static bool try_to_update_entry(const char *name, const char *value, const size_t len)
+{
+	if (!name || len == 0 || !value) {
+		return false;
+	}
+
+	struct settings_ram_entry *entry = NULL;
+	SYS_SLIST_FOR_EACH_CONTAINER (&settings_list, entry, node) {
+		if (entry) {
+			if (memcmp(entry->key, name, entry->key_len) == 0) {
+				/* first free old memory */
+				k_free(entry->value);
+				entry->value = k_malloc(len);
+				if (entry->value) {
+					memcpy(entry->value, value, len);
+				}
+				entry->value_len = len;
+				return true;
+			}
+		}
+	}
+	return false;
+}
+
+/**
+ * @brief try to get an entry from the settings list
+ * 
+ * Searches along the settings list and 
+ * if the entry is found update value and len arguments.
+ * 
+ * @param name name of the entry
+ * @param value pointer to memory where the entry's value will be stored
+ * @param len length of the read out entry
+ * @return true if entry has been found and read out
+ * @return false if could not find given entry
+ */
+static bool get_entry(const char *name, char *value, size_t *len)
+{
+	if (!name || !value || !len) {
+		return false;
+	}
+	struct settings_ram_entry *entry = NULL;
+	SYS_SLIST_FOR_EACH_CONTAINER (&settings_list, entry, node) {
+		if (entry) {
+			if (memcmp(entry->key, name, entry->key_len) == 0) {
+				memcpy(value, entry->value, entry->value_len);
+				*len = entry->value_len;
+#ifdef CONFIG_SETTINGS_RAM_DEBUG
+				LOG_PRINTK("Reading setting: %s with size: %zu : [", name, *len);
+				for (size_t i = 0; i < *len; i++) {
+					LOG_PRINTK("0x%" PRIx8 ", ", value[i]);
+				}
+				LOG_PRINTK("]\n");
+				LOG_PRINTK("\n");
+#endif
+				return true;
+			}
+		}
+	}
+	return false;
+}
+
+/**
+ * @brief Delete existing entry
+ * 
+ * Search along the settings list to find the given entry,
+ * if the entry is found, free memory of the entry's value 
+ * and free memory for the full entry.
+ * 
+ * @param name entry name to be deleted
+ * @return true if entry has been deleted
+ * @return false entry has not been found
+ */
+static bool delete_entry(const char *name)
+{
+	struct settings_ram_entry *entry = NULL;
+	bool rc = false;
+	SYS_SLIST_FOR_EACH_CONTAINER (&settings_list, entry, node) {
+		if (entry) {
+			if (memcmp(entry->key, name, entry->key_len) == 0) {
```

Comment:
Can the `entry` variable ever be NULL? Either way, these conditions could be checked in a single statement.

```suggestion
		if (entry != NULL && memcmp(entry->key, name, entry->key_len) == 0) {
```